modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py

```diff
@@ -462,6 +462,34 @@ def applier(lib):
 
         eval_general(pd, pandas, applier)
 
+    @pytest.mark.parametrize("is_multiindex", [True, False])
+    @pytest.mark.parametrize(
+        "column_names", [None, ["level1", None], ["level1", "level2"]]
+    )
+    def test_reset_index_multicolumns(self, is_multiindex, column_names):
+        index = (
+            pandas.MultiIndex.from_tuples(
+                [(i, j, k) for i in range(2) for j in range(3) for k in range(4)],
+                names=["l1", "l2", "l3"],
+            )
+            if is_multiindex
+            else pandas.Index(np.arange(len(self.data["a"])), name="index")
+        )
+        columns = pandas.MultiIndex.from_tuples(
+            [("a", "b"), ("b", "c")], names=column_names
+        )
+        data = np.array(list(self.data.values())).T
+
+        def applier(df, **kwargs):
+            df = df + 1
```

Comment:
HACK: implicit OmniSci' execution triggering, the same as it's done [everywhere else](https://github.com/modin-project/modin/blob/f9c99c3817d35f7bf0d91df9aa9644545ea27517/modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py#L461) at this file